src/views/NewQuestion/Sections/SectionForm.tsx

```diff
@@ -24,30 +29,146 @@ const defaultOptions = {
   const [studentGrade, setGrade] = useState(defaultOptions as Option);
   const [isPublic, setIsPublic] = useState(true as boolean);
+  const [azureToken, setAzureToken] = useState('' as string);
+  const [tempFiles, setTempFiles] = useState([] as any[]);
 
   useEffect(() => {
     getSubjectList().then(setSubjects);
   }, []);
 
-  const handleSubmit = () => {
-    const questionForm: IQuestion = {
-      email,
-      studentGrade: Number(studentGrade.value),
-      subjectID: Number(subject.value),
-      themeID: Number(theme.value),
-      questionText,
-      isPublic,
-      totalRows: 0,
+  useEffect(() => {
+    window.sessionStorage.clear();
+    const localAzureToken = window.sessionStorage.getItem('azuretoken');
+    const generatedToken = {
+      token: localAzureToken
+        ? JSON.parse(localAzureToken).token
+        : jws.sign({
+            header: { alg: 'HS256' },
+            payload: 'Questionfiles',
+            secret: secureRandom(256, { type: 'Buffer' }),
+          }),
     };
-    postQuestion(questionForm).then(res => console.log(res));
+    window.sessionStorage.setItem('azuretoken', JSON.stringify(generatedToken));
+    setAzureToken(generatedToken.token);
+  }, []);
+
+  /** KEEP TO USE IN FRIVILLIG APP TO HANDLE FILE AND DIRECTORY DELETION
+  const handleDirectoryDelete = () => {
+    console.log('Kjører frem til if check');
+    files.length > 0 &&
+      azureToken.length > 0 &&
+      files.map((file, index) => {
+        console.log(file);
+        fileService.deleteFileIfExists(
+          'questionfiles',
+          azureToken,
+          file.fileName,
+          function(error, response) {
+            if (!error) {
+            }
+          },
+        );
+        if (index == files.length - 1) {
+          setFiles([] as IFile[]);
+          fileService.deleteDirectoryIfExists(
+            'questionfiles',
+            azureToken,
+            function(error, result, response) {
+              if (!error) {
+                console.log(result);
+                console.log(response);
+              }
+            },
+          );
+        }
+      });
+  };*/
+
+  const uploadPromises = tempFiles => {
+    return tempFiles.map(async file => {
+      return uploadFileToAzureFileStorage(
+        'questionfiles',
+        azureToken,
+        file,
+        azureToken,
+      );
+    });
+  };
+
+  const handleSubmit = () => {
+    return Promise.all<IFile>(uploadPromises(tempFiles)).then(results => {
+      const questionForm: IQuestion = {
+        email,
+        studentGrade: Number(studentGrade.value),
+        subjectID: Number(subject.value),
+        themeID: Number(theme.value),
+        questionText,
+        isPublic,
+        totalRows: 0,
+        files: results,
+      };
+      postQuestion(questionForm).then(() => {
+        history.push({ pathname: '/questions/new/success' });
+      });
+    });
+  };
+
+  function MyDropzone() {
+    const onDrop = useCallback(acceptedFiles => {
```

Comment:
User has to be able to upload multiple files at once

Reply:
Godt observert! Fikset :)